### PR TITLE
DSND:733 - Added integration tests

### PR DESCRIPTION
**Pull request by vchitti-kainos**

### CODE REVIEW

File: src/itest/java/uk/gov/companieshouse/company/links/steps/CompanyLinksSteps.java [hunk 1]
```diff
@@ -85,15 +106,45 @@ public void the_company_links_consumer_should_send_a_patch_request_to_the_compan
         wireMockServer.stop();
     }
 
+    @When("a non-avro message is published and failed to process")
+    public void a_non_avro_message_is_published_and_failed_to_process() throws InterruptedException {
+        configureWiremock();
+        kafkaTemplate.send(topic, "invalid message");
+
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        countDownLatch.await(5, TimeUnit.SECONDS);
```
Comment (amobarak) [inline src/itest/java/uk/gov/companieshouse/company/links/steps/CompanyLinksSteps.java:115]: Configure the second wait time as a constant in the top of CompanyLinksSteps

**Review by amobarak — APPROVED** [review]
Looks good, just one small comment

File: src/itest/resources/features/company-links.feature [hunk 2]
```diff
@@ -1,13 +1,45 @@
...
     Given Company links consumer api service is running
-    When a message is published for companyNumber "00006400" to check for links
+    When a message is published for companyNumber "00006400" to check for links with status code "200"
     Then the Company Links Consumer should send a GET request to the Company Profile API
+
+  Scenario: Consume invalid message
+
+    Given Company links consumer api service is running
+    When a non-avro message is published and failed to process
+    Then the message should be moved to topic "stream-company-insolvency-company-links-consumer-invalid"
```
Comment (zahidm3) [inline src/itest/resources/features/company-links.feature:19]: can you please separate happy path and error-retry stories into separate feature files?

**Review by zahidm3 — CHANGES_REQUESTED** [review]

File: src/itest/resources/features/company-links.feature [hunk 3]
```diff
@@ -1,13 +1,45 @@
 Feature: Process company links information
 
-  Scenario: Consume the message and creating the company links
+  Scenario: Consume the message and creating the company links successfully
 
     Given Company links consumer api service is running
     When a message is published for companyNumber "00006401" to update links
     Then the Company Links Consumer should send a PATCH request to the Company Profile API
```
[the remark this answers is not in the document]
Comment (vchitti-kainos) [reply src/itest/resources/features/company-links.feature:7]: its part of another story.

**Review by zahidm3 — APPROVED** [review]

**Pull request merged**